File: Backend/ServerBackend/src/main/java/com/fran/dev/potjera/serverbackend/models/gamesession/GameSessionState.java
package com.fran.dev.potjera.serverbackend.models.gamesession;

import com.fran.dev.potjera.serverbackend.models.gamesession.coinbooster.CoinBoosterPlayerState;
import com.fran.dev.potjera.serverbackend.models.gamesession.playervhunter.PlayerVHunterBoardState;
import com.fran.dev.potjera.serverbackend.models.gamesession.playervhunter.PlayerVHunterGlobalState;
import lombok.Builder;
import lombok.Data;

import java.util.Map;

@Data
@Builder
public class GameSessionState {
    private String gameSessionId;
    private Map<Long, CoinBoosterPlayerState> coinBoosterPlayerStates;
    private PlayerVHunterGlobalState playerVHunterGlobalState;
    private GameSessionStage gameSessionStage;
    private Map<Long, PlayerVHunterBoardState> playerBoardStates;

    public void updateCoinBoosterPlayerState(CoinBoosterPlayerState newPlayerState) {
        coinBoosterPlayerStates.put(newPlayerState.getPlayerId(), newPlayerState);
    }
}

File: AndroidApp/app/src/main/java/com/fran/dev/potjera/android/app/game/models/state/GameSessionState.kt
package com.fran.dev.potjera.android.app.game.models.state

import com.fran.dev.potjera.android.app.game.models.GameSessionPlayer
import com.fran.dev.potjera.android.app.game.models.enums.GamePhase

data class GameSessionState(
    val gameSessionId: String = "",
    val gamePhase: GamePhase = GamePhase.STARTING,
    val gameSessionPlayers: Map<Long, GameSessionPlayer> = emptyMap(),
)